src/components/Layout/Header/NavBar/ChainMenu.tsx

```diff
@@ -51,19 +58,60 @@ type ChainMenuProps = BoxProps
 
 export const ChainMenu = (props: ChainMenuProps) => {
   const { state, load } = useWallet()
-  const { isLoading, supportedEvmChainIds, connectedEvmChainId, setEthNetwork } = useEvm()
+  const {
+    connectedEvmChainId,
+    getChainIdFromEthNetwork,
+    isLoading,
+    setEthNetwork,
+    supportedEvmChainIds,
+  } = useEvm()
   const chainAdapterManager = getChainAdapterManager()
   const translate = useTranslate()
 
-  const handleChainClick = async (chainId: ChainId) => {
-    try {
-      await (state.wallet as ETHWallet).ethSwitchChain?.(Number(chainId))
-      setEthNetwork(chainId)
-      load()
-    } catch (e) {
-      // TODO: Handle me after https://github.com/shapeshift/hdwallet/pull/551 is published
-    }
-  }
+  const handleChainClick = useCallback(
+    async (requestedEthNetwork: string) => {
+      try {
+        const requestedChainId = getChainIdFromEthNetwork(requestedEthNetwork)
+
+        if (!requestedChainId) {
+          throw new Error(`Unsupported EVM network: ${requestedEthNetwork}`)
+        }
+
+        const requestedChainChainAdapter = chainAdapterManager.get(requestedChainId) as unknown as
+          | EvmBaseAdapter<EvmChainId>
+          | undefined
+
+        if (!requestedChainChainAdapter) {
+          throw new Error(`No chain adapter found for: ${requestedChainId}`)
+        }
+
+        const requestedChainFeeAssetId = requestedChainChainAdapter.getFeeAssetId()
+        const requestedChainFeeAsset = selectAssetById(store.getState(), requestedChainFeeAssetId)
```

Comment:
Weird to use `store.getState()` within a component, but that's a lot cleaner than other options, since this is a memoized callback and we can't  really set state fields and have them as dependencies, which means we would need to fire it again

Reply:
just pass assets in as a dep and index by key?